axum-extra/src/routing/mod.rs

```diff
@@ -30,12 +30,12 @@ pub use self::typed::{SecondElementIs, TypedPath};
 #[doc(hidden)]
 #[must_use]
 pub const fn __private_validate_static_path(path: &'static str) -> &'static str {
-    if path.is_empty() {
-        panic!("Paths must start with a `/`. Use \"/\" for root routes")
-    }
-    if path.as_bytes()[0] != b'/' {
-        panic!("Paths must start with /");
-    }
+    assert!(
+        !path.is_empty(),
+        "Paths must start with a `/`. Use \"/\" for root routes"
+    );
+    // `assert_eq!` is not allowed in constant functions so we have to do it manually.
```

Comment:
I think I commented on this before, but I wouldn't want `assert_eq` here even if it was possible, since it adds a LHS-RHS comparison to the output, which would not be helpful here.